mlflow/gateway/providers/mosaicml.py

```diff
@@ -0,0 +1,109 @@
+from typing import Dict, Any
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import send_request, rename_payload_keys
+from mlflow.gateway.schemas import chat, completions, embeddings
+from mlflow.gateway.config import MosaicMLConfig, RouteConfig
+
+
+class MosaicMLProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(config.model.config, MosaicMLConfig):
+            raise TypeError(f"Unexpected config type {config.model.config}")
+        self.mosaicml_config: MosaicMLConfig = config.model.config
+
+    async def _request(self, model: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        headers = {"Authorization": f"{self.mosaicml_config.mosaicml_api_key}"}
+        return await send_request(
+            headers=headers,
+            base_url="https://models.hosted-on.mosaicml.hosting",
```

Comment:
Should we make this configurable? Seems like there's a good chance this could change in the medium term :). Using `"https://models.hosted-on.mosaicml.hosting"` as a default makes sense, but users should be able to override

Reply:
How would we want users to be able to override this? An additional parameter in `create_route` or in the payload when `querying` each time?

Reply:
The config can be updated with an override entry for processing from the config.yaml :) I'll add this to the commit I'm going to make in a few minutes.... one sec :)